homeassistant/auth/__init__.py

```diff
@@ -158,6 +160,11 @@ def __init__(
         self._mfa_modules = mfa_modules
         self.login_flow = AuthManagerFlowManager(hass, self)
         self._revoke_callbacks: dict[str, list[CALLBACK_TYPE]] = {}
+        self._expire_callbacks: dict[str, CALLBACK_TYPE] = {}
+        self._expire_callback: CALLBACK_TYPE | None = None
+
+        hass.async_run_job(self.async_track_next_refresh_token_expiration)
+        hass.async_add_shutdown_job(HassJob(self.async_cancel_expiration_schedule))
```

Comment:
It would be nicer if this happened in a `async_setup`

Reply:
```diff
diff --git a/homeassistant/auth/__init__.py b/homeassistant/auth/__init__.py
index e160a45db25..36338d54533 100644
--- a/homeassistant/auth/__init__.py
+++ b/homeassistant/auth/__init__.py
@@ -76,7 +76,9 @@ async def auth_manager_from_config(
     for module in modules:
         module_hash[module.id] = module
 
-    return AuthManager(hass, store, provider_hash, module_hash)
+    manager = AuthManager(hass, store, provider_hash, module_hash)
+    manager.async_setup()
+    return manager
 
 
 class AuthManagerFlowManager(data_entry_flow.FlowManager):
@@ -162,9 +164,14 @@ class AuthManager:
         self._revoke_callbacks: dict[str, list[CALLBACK_TYPE]] = {}
         self._expire_callbacks: dict[str, CALLBACK_TYPE] = {}
         self._expire_callback: CALLBACK_TYPE | None = None
+        self._remove_expired_job = HassJob(self._async_remove_expired_refresh_tokens)
 
-        hass.async_add_job(self.async_track_next_refresh_token_expiration)
+    @callback
+    def async_setup(self) -> None:
+        """Set up the auth manager."""
+        hass = self.hass
         hass.async_add_shutdown_job(HassJob(self.async_cancel_expiration_schedule))
+        self.async_track_next_refresh_token_expiration()
 
     @property
     def auth_providers(self) -> list[AuthProvider]:
@@ -513,7 +520,7 @@ class AuthManager:
                 next_expiration = expire_at
 
         self._expire_callback = async_track_point_in_utc_time(
-            self.hass, self._async_remove_expired_refresh_tokens, next_expiration
+            self.hass, self._remove_expired_job, next_expiration
         )
 
     async def async_cancel_expiration_schedule(self) -> None:

```

Reply:
thanks @bdraco for this great improvement -> [move scheduler init to async_setup 3350077](https://github.com/home-assistant/core/pull/108428/commits/335007754adaabfade1b64c89a0fdb72f7e7c76a)